From: Stefan Bader <stefan.bader@canonical.com>
Date: Wed, 27 May 2015 14:57:05 +0200
Subject: ubuntu: libxl: Set common qemu path

Upstream Xen assumes that qemu-system-x86 is shipped as part of the Xen
package. However, since this is now merged back into common qemu and
we (and Debian) do not ship an additional binary, the binary is not
found under the LIBXL_EXECBIN_DIR. So instead, refer to the common
path (/usr/bin).

BugLink: http://bugs.launchpad.net/bugs/1459600

Signed-off-by: Stefan Bader <stefan.bader@canonical.com>

Index: ubuntu-libvirt/src/libxl/libxl_conf.c
===================================================================
--- ubuntu-libvirt.orig/src/libxl/libxl_conf.c
+++ ubuntu-libvirt/src/libxl/libxl_conf.c
@@ -435,7 +435,7 @@ libxlCapsInitGuests(libxl_ctx *ctx, virC
         if ((guest = virCapabilitiesAddGuest(caps,
                                              guest_archs[i].hvm ? VIR_DOMAIN_OSTYPE_HVM : VIR_DOMAIN_OSTYPE_XEN,
                                              guest_archs[i].arch,
-                                             LIBXL_EXECBIN_DIR "/qemu-system-i386",
+                                             "/usr/bin/qemu-system-i386",
                                              (guest_archs[i].hvm ?
                                               LIBXL_FIRMWARE_DIR "/hvmloader" :
                                               NULL),
Index: ubuntu-libvirt/src/libxl/libxl_domain.c
===================================================================
--- ubuntu-libvirt.orig/src/libxl/libxl_domain.c
+++ ubuntu-libvirt/src/libxl/libxl_domain.c
@@ -371,6 +371,18 @@ libxlDomainDefPostParse(virDomainDefPtr
     if (virDomainDefCheckUnsupportedMemoryHotplug(def) < 0)
         return -1;
 
+    /*
+     * For older versions of libvirt we always set the emulator string
+     * to "qemu-dm" (the field was not actually used or passed on to
+     * libxl). Newer versions of libvirt do use it and also pass it on.
+     * Also xml verification insists on this being a full path. So we
+     * convert all old configs to the full path.
+     */
+    if (def->emulator && STREQ(def->emulator, "qemu-dm")) {
+        VIR_FREE(def->emulator);
+        if (VIR_STRDUP(def->emulator, "/usr/bin/qemu-system-i386") < 0)
+            return -1;
+    }
     return 0;
 }
 
